test2code-admin/src/main/kotlin/com/epam/drill/plugins/test2code/storage/Session.kt

```diff
@@ -36,30 +38,43 @@ data class AgentKey(
 internal class StoredSession(
     @Id val id: String,
     val agentKey: AgentKey,
-    val scopeId: String,
-    val data: FinishedSession,
+    @Id val sessionHolderId: String,
+    val testType: String,
+    val testName: String,
+    val tests: Set<TestOverview>,
+    var probes: List<ExecClassData>,
 )
 
 internal suspend fun StoreClient.loadSessions(
-    scopeId: String,
-): List<FinishedSession> = trackTime("Load session") {
+    holderId: String,
+): List<TestSession> = trackTime("Load session") {
     findBy<StoredSession> {
-        StoredSession::scopeId eq scopeId
-    }.get().map { it.data }
+        StoredSession::sessionHolderId eq holderId
+    }.get().map { stored ->
+        TestSession(
+            id = stored.id,
+            testType = stored.testType,
+            testName = stored.testName,
+        ).also { session ->
```

Comment:
Please do not use `also` to manipulate data -> instead use it _only_ for side-effects (for things like I/O, logging etc)
[`apply`](https://kotlinlang.org/docs/scope-functions.html#apply) would work better in this case

Why:
- [`also`](https://kotlinlang.org/docs/scope-functions.html#also) implies no changes to the object even judging by it's documentation - "...Use also for actions that need a reference to the object rather than its properties and functions"
- `apply` on other hand is clearly purposed for object modification - "...The most common use case for apply is for object configuration. Such calls can be read as "apply the following assignments to the object.""

Reply:
chanhed to apply